src/plugins/intel_cpu/src/nodes/executors/dnnl/dnnl_fullyconnected_primitive.cpp

```diff
@@ -228,6 +228,9 @@ static dnnl::inner_product_forward::primitive_desc createDescriptorInternal(cons
         wdt = weightDesc.get_data_type();
     } else if (indt == dnnl::memory::data_type::u8 || indt == dnnl::memory::data_type::s8) {
         wdt = memory::data_type::s8;
+    } else if (indt == memory::data_type::f32 && weightDesc.get_data_type() == memory::data_type::f16) {
```

Comment:
1. We also should check that avx2 is available on the system (requirement for vcvtph2ps)
2. I would propose to update useWeightsDecompressionImpl instread of custom condition here which is also required to filter out unsupported implementations in https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/nodes/executors/fullyconnected_implementations.cpp#L157

Reply:
OK